tests/attacks/poison/test_sleeper_agent_attack.py

```diff
@@ -0,0 +1,148 @@
+# MIT License
+#
+# Copyright (C) The Adversarial Robustness Toolbox (ART) Authors 2022
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+# documentation files (the "Software"), to deal in the Software without restriction, including without limitation the
+# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+# persons to whom the Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
+# Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+# WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+# TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import logging
+import numpy as np
+import pytest
+from PIL import Image
+
+from art.attacks.poisoning.sleeper_agent_attack import SleeperAgentAttack
+from art.utils import to_categorical
+from numpy import asarray
+
+from tests.utils import ARTTestException
+
+logger = logging.getLogger(__name__)
+
+
+
+@pytest.mark.only_with_platform("pytorch", "tensorflow2")
+def test_poison(art_warning, get_default_mnist_subset, image_dl_estimator,framework):
+    try:
+        (x_train, y_train), (x_test, y_test) = get_default_mnist_subset
+        if framework == 'pytorch':
+            x_train = np.transpose(x_train, (0, 2, 3, 1))
```

Comment:
Can we test PyTorch with channels first images?

Reply:
Added Functionality provided for handling both channel first and channel last